src/public/app/room/PrivateQueue.ts

```diff
@@ -0,0 +1,53 @@
+/* import { Observable } from 'rxjs/Observable';
+import { Observer } from 'rxjs/Observer'; */
+import { Queue, QueuedUpSong } from '../core/types';
+
+
+class PrivateQueue implements Queue {
+
+    private songs: QueuedUpSong[] = [];
+    /*
+    private songsObservable: Observable<QueuedUpSong[]>;
+    private songsObserver: Observer<QueuedUpSong[]>;
+    */
+    constructor() {
+        /*
+        this.songsObservable = Observable.create( (observer: Observer<QueuedUpSong[]>) => {
+           this.songsObserver = observer; 
+        });
+        */
+    }
+
+
+    public addSong(song: QueuedUpSong): void {
+        this.songs.push(song);
+       // this.songsObserver.next(this.songs);
+    }
+
+    public nextSong(): QueuedUpSong {
+        if (!this.songs.length) {
+            return null;
+        }
+        let nextSong: QueuedUpSong = this.songs.shift();
+        // this.songsObserver.next(this.songs);
+        return nextSong;
+    }
+
+    public removeSongAtIndex(index: number): void {
+        this.songs = this.songs.filter( (song: QueuedUpSong, i: number) => {
```

Comment:
I did this because angular2 recommends immutable data structures. It helps with change detection to have to store only the pointer to the main object rather than having to figure out if something within the data structure changed.